cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.cpp

```diff
@@ -129,11 +98,34 @@ std::shared_ptr<StandardHonkComposerHelper::VerificationKey> StandardHonkCompose
     if (verification_key) {
         return verification_key;
     }
+    // TODO(luke): These checks for proving_key/commitment_key seem to only be used when we're testing
+    // compute_verification_key and dont want to be bothered to explicitly compute the proving_key/commitment_key
+    // externally. This seems like a bad model and leads to clutter. Should these just become asserts?
     if (!proving_key) {
         compute_proving_key(circuit_constructor);
     }
+    if (!commitment_key) {
+        commitment_key = std::make_shared<PCSParams::CommitmentKey>(proving_key->circuit_size, crs_factory_);
+    }
+
+    verification_key = std::make_shared<VerificationKey>(
+        proving_key->circuit_size, proving_key->num_public_inputs, proving_key->composer_type);
```

Comment:
why is the composer_type needed here?

Reply:
For now that's how the constructor is defined. It may be that it can be removed. Eventually the composer type is going to be removed altogether so that would be a natural time to evaluate